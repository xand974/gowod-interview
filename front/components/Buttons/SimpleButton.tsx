import { StyleSheet } from "react-native";
import styled from "styled-components/native";

const Button = styled.TouchableOpacity`
  padding: 20px 10px;
  background: ${(props: { bg: string }) => props.bg};
  width: 100%;
  align-items: center;
  justify-content: center;
  border-radius: 10px;
`;

const ButtonText = styled.Text`
  color: white;
  font-weight: 600;
`;

interface SimpleButtonProp {
  text: string;
  bg: string;
  onPress: () => void;
}
export default function SimpleButton({ text, bg, onPress }: SimpleButtonProp) {
  return (
    <Button onPress={onPress} bg={bg}>
      <ButtonText>{text}</ButtonText>
    </Button>
  );
}

const styles = StyleSheet.create({});
