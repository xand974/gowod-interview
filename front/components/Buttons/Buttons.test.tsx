import SimpleButton from "./SimpleButton";
import renderer from "react-test-renderer";

test("render Simple Button component", () => {
  const component = renderer.create(
    <SimpleButton
      text="Sample"
      bg="red"
      onPress={() => {
        return true;
      }}
    ></SimpleButton>
  );
  const componentJSON = component.toJSON() as any;
  const element = component.toTree();
  const renderedElement = element.rendered as any;

  expect(componentJSON.props.style.backgroundColor).toEqual("red");
  expect(renderedElement?.props.onPress()).toEqual(true);
  expect(componentJSON?.children?.length).toEqual(1);
  expect(componentJSON).toMatchSnapshot();
});
