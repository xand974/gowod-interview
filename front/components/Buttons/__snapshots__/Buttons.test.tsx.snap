// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Simple Button component 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "red",
      "borderRadius": 10,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 20,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 20,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontWeight": "600",
        },
      ]
    }
  >
    Sample
  </Text>
</View>
`;
